source/plugin/event/causes.rst

```diff
@@ -69,55 +99,164 @@ object matching the provided type.
+as a saved state, when you are done with your causes, the removal of the frame returns the manager to
+its original state.
+
+.. tip::
+
+    Adding a frame to the CauseStackManager does not remove what is already in the manager, so anything
+    that is in the cause stack and contexts before a stack frame is added will be there afterwards. You
+    can verify this by calling ``Sponge.getCauseStackManager().getCurrentCause()`` before and after the
+    frame is pushed.
+
+    For example, if the cause stack contains a ``PluginContainer`` and a ``CommandSource`` when a frame
+    is pushed, they will remain on the stack and will form part of the ``Cause`` if one is obtained from
+    the frame.
+
+For example, if you were to fire an event that was simulating another player in a sudo like command,
+you may want to add your acting player in the cause and the game profile of the player that you are
+simulating in the context (as the simulated player is not directly responsible for the event being fired.)
+
+**Creating a custom Cause with the CauseStackManager**
+
+In this example, the variables would be populated, the cause would contain the ``playerToSimulate`` as
+the root cause, the ``sourceRunningSudo`` as the second object in the cause and the :javadoc:`GameProfile`
+as the :javadoc:`EventContextKeys#SIMULATED_PLAYER` context, in addition to anything already in the
+``CauseStackManager``. Your event code would be at the bottom of the method.
 
-If you already have a cause object and would like to append some more objects to the chain you can use
-:javadoc:`Cause#with(NamedCause, NamedCause...)`. This constructs a new Cause object containing first the objects
-already present in the original cause, then followed by the additional objects that you provided.
+.. code-block:: java
+
+    // In your code these would be populated
+    CommandSource sourceRunningSudo;
+    Player playerToSimulate;
+    try (CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {
+      // This would be the player that is causing the action (would have run the /sudo command)
+      frame.pushCause(sourceRunningSudo);
+
+      // You may want to push the simulated player so that it's at the root for compatibility
+      // with other plugins that listen for the direct cause
+      frame.pushCause(playerToSimulate);
+
+      // This would contain the GameProfile of the player to simulate, they are not a direct
+      // cause, but provide additional information to the event listeners so that they know
+      // the root player is a simulated one and may not have invoked the event.
+      frame.addContext(EventContextKeys.SIMULATED_PLAYER, playerToSimulate.getProfile());
+
+      // Add your event code here.
+      // You can get the current cause at any time using the getCurrentCause method
+      // (which is available on the fram as well as the CauseStackManager)
```

Comment:
I will concur with the observation that you put a lot of comments in the code blocks (compared to existing docs). These cannot be translated by _Crowdin_ (it sensibly ignores code blocks), so it does limit their reach somewhat. We prefer comments were left *outside* the code blocks wherever possible, for a host of reasons. There _are_ numerous good reasons to have comments in there; are you sure this is one of those cases?